src/main/scala/net/artsy/atomic/AtomicEventStore.scala

```diff
@@ -203,32 +205,25 @@ abstract class AtomicEventStore[EventType <: Serializable: Scoped: ClassTag, Val
   class Receptionist(
     logProps: String => Props
   ) extends Actor {
-    var logs = Map.empty[String, ActorRef]
+    val logs = scala.collection.mutable.Map.empty[String, ActorRef]
 
-    def liveLogForScope(scope: String): ActorRef = {
-      val (newLogs, targetLog) = logs.get(scope) match {
-        case Some(foundLog) => (logs, foundLog)
-        case None =>
-          // Recreate the log, which will recall all preexisting events
-          val materializedLog = context.actorOf(logProps(scope), scope)
+    def liveLogForScope(scope: String): ActorRef = logs.getOrElseUpdate(scope, createLog(scope))
```

Comment:
I think it's easier to understand if we don't chain like 

`logs.getOrElseUpdate(scope, createLog(scope)).forward(message)`

I don't have a problem with having really small functions.